addon/webextension/background/analytics.js

```diff
@@ -8,34 +8,70 @@ this.analytics = (function() {
   let telemetryPrefKnown = false;
   let telemetryPref;
 
+  let events = [];
+  let timings = [];
+  let eventsIdleCallbackHandle, timingsIdleCallbackHandle;
+  const fetchOptions = {
+    method: "POST",
+    mode: "cors",
+    headers: { "content-type": "application/json" },
+    credentials: "include"
+  };
+
+  function flushEvents() {
+    if (events.length === 0) {
+      return;
+    }
+
+    let eventsUrl = `${main.getBackend()}/event`;
+    let deviceId = auth.getDeviceId();
+    let sendTime = Date.now();
+
+    events.forEach(event => {
+      event.queueTime = sendTime - event.eventTime
+      log.info(`sendEvent ${event.event}/${event.action}/${event.label || 'none'} ${JSON.stringify(event.options)}`);
+    });
+
+    let body = JSON.stringify({deviceId, events});
+    let fetchRequest = fetch(eventsUrl, Object.assign({body}, fetchOptions));
+    fetchWatcher(fetchRequest);
+
+    events = [];
+    eventsIdleCallbackHandle = null;
```

Comment:
I don't think there's any good way for this to happen, but potentially if `events` is empty then the handle won't be nulled out. I'd be inclined to reset `eventsIdleCallbackHandle` in the same place it is initialized. But as a separate issue it's not initialized.